orchagent/aclorch.cpp

```diff
@@ -4660,16 +5004,43 @@ void AclOrch::doAclRuleTask(Consumer &consumer)
             }
 
             if (bHasIPV4 && bHasIPV6)
-	    {
-		    if (type == TABLE_TYPE_L3V4V6)
-		    {
-			    SWSS_LOG_ERROR("Rule '%s' is invalid since it has both v4 and v6 matchfields.", rule_id.c_str());
-			    bAllAttributesOk = false;
-		    }
-	    }
+	        {
+		        if (type == TABLE_TYPE_L3V4V6)
+		        {
+			        SWSS_LOG_ERROR("Rule '%s' is invalid since it has both v4 and v6 matchfields.", rule_id.c_str());
+			        bAllAttributesOk = false;
+		        }
+	        }
+            bool egrDscpRuleStatus = true;
+            if (needsEgrSetDscpRule)
+            {
+                auto metadata = m_egrDscpRuleMetadata[key];
+
+                if (m_metadataEgrDscpRule[metadata].size() == 1)
+                {
+                    // Create EGR_SET_DSCP rule
+                    auto egrSetDscpRule = make_shared<AclRuleUnderlaySetDhcp>(this, std::to_string(metadata), EGR_SET_DSCP_TABLE_ID);
+                    egrSetDscpRule->validateAddMatch(MATCH_METADATA, std::to_string(metadata));
+                    egrSetDscpRule->validateAddAction(ACTION_DSCP, std::to_string(actionDscpValue));
+
+                    if (egrSetDscpRule->validate())
+                    {
+                        if (!addAclRule(egrSetDscpRule, EGR_SET_DSCP_TABLE_ID))
+                        {
+                            SWSS_LOG_ERROR("Failed to create ACL rule %d in table %s", metadata, EGR_SET_DSCP_TABLE_ID);
+                            egrDscpRuleStatus = false;
+                        }
+                    }
+                    else
+                    {
+                        SWSS_LOG_ERROR("Failed to validate ACL rule %d in table %s", metadata, EGR_SET_DSCP_TABLE_ID);
+                        egrDscpRuleStatus = false;
+                    }
+                }
+            }
 
             // validate and create ACL rule
-            if (bAllAttributesOk && newRule->validate())
+            if (egrDscpRuleStatus && bAllAttributesOk && newRule->validate())
```

Comment:
Why check `egrDscpRuleStatus` here?

Reply:
same logic as before.
egrDscpRuleStatus = true
if table type is UNDERLAY_SET_DSCP
then create EGR_SET_DSCP rule . failure should set egrDscpRuleStatus =false
if egrDscpRuleStatus is false then dont create other rule.
if egrDscpRuleStatus is true then create MARK_META/v6 rule or other table rule for which the request came in.